pvlib/modelchain.py

```diff
@@ -1271,6 +1291,17 @@ def _verify(data, index=None):
             for (i, array_data) in enumerate(data):
                 _verify(array_data, i)
 
+    def _configure_results(self):
```

Comment:
an alternative to this method might be to instantiate `ModelChain` with `self.results = None` (or not at all), then put this line at the start of each `run_model` workflow: 

```
self.results = ModelChainResults(singleton_tuples=(self.system.num_arrays == 1 and isinstance(self.weather, tuple))
```

I guess maybe you'd still want to extract that logic into a method like this one. Then the difference comes down to setting the right value vs instantiating the object with the right value. 🤷

Reply:
This would basically be a transformation from `_configure_results` to `_prepare_results`. I kind of like that better, but I don't think it makes a big difference. It would also take a little bit of work, since some of the tests make the (reasonable) assumption that all the results fields exist before one of the `run_model` workflows is triggered (e.g. `test__assign_total_irrad` and `ModelChain.complete_irradiance`).

Reply:
Should `_configure_results()` (or some `ModelChainResult` method) iterate over per-array fields to and wrap them in tuple if `_singleton_tuples` is changed from False to True?

Reply:
I'm viewing ModelChainResult as a single use object so I don't think we need to watch for that kind of state change.